quic/s2n-quic-transport/src/contexts/mod.rs

```diff
@@ -28,6 +28,23 @@ pub trait WriteContext {
     /// Returns the number of available bytes remaining in the current payload
     fn remaining_capacity(&self) -> usize;
 
+    /// Attempt to write an ack frame.
+    ///
+    /// If this was successful the number of the packet
+    /// that will be used to send the frame will be returned.
+    fn write_ack_frame<Frame, AckRanges: AckRangesTrait>(
```

Comment:
Perhaps we should not have a blanket implementation, and implement this dummy version of `write_ack_frame` in `MockContext` instead?

Reply:
A default implementation is nice to have, especially if you're not testing the events. 

That being said, why not just take an actual `Ack` frame and remove the `ack_ranges` parameter? I don't think we really get anything by being generic over the actual ack frame here.

Reply:
done